static/js/lib/learning_resources.js

```diff
@@ -27,24 +32,130 @@ export const availabilityLabel = (availability: ?string) => {
   }
 }
 
+export const availabilityFilterToMoment = (filter: string, ending: boolean) => {
+  // Convert an Elasticsearch date_range filter string to a moment
+  const format = /(now)(\+)?(\d+)?([Md])?/
+  const match = format.exec(filter)
+  if (match) {
+    let dt = moment()
+    if (match[3] && match[4]) {
+      dt = dt.add(match[3], match[4] === "d" ? "days" : "months")
+    }
+    if (ending) {
+      // $FlowFixMe: this is fine according to moment docs, and it works
+      dt.set({ hour: 23, minute: 59, second: 59 })
+    } else {
+      // $FlowFixMe: this is fine according to moment docs, and it works
+      dt.set({ hour: 0, minute: 0, second: 0 })
+    }
+    return dt
+  }
+}
+
+export const inDateRanges = (run: CourseRun, availabilities: Array<string>) => {
+  let inRange = false
+  availabilities.forEach(availability => {
+    if (AVAILABILITY_MAPPING[availability]) {
+      const from = availabilityFilterToMoment(
+        AVAILABILITY_MAPPING[availability].filter.from,
+        false
+      )
+      const to = availabilityFilterToMoment(
+        AVAILABILITY_MAPPING[availability].filter.to,
+        true
+      )
+      const startDate = runStartDate(run)
+      if (
+        (isNil(from) ||
+          // $FlowFixMe: if we get this far, only moments are compared
+          startDate.isSameOrAfter(from)) &&
+        (isNil(to) ||
+          // $FlowFixMe: if we get this far, only moments are compared
+          startDate.isSameOrBefore(to))
+      ) {
+        inRange = true
+      }
+    }
+  })
+  return inRange
+}
+
+export const bestRunLabel = (run: ?CourseRun) => {
+  if (!run) {
+    return AVAILABILITY_MAPPING[AVAILABLE_NOW].label
+  }
+  for (const range in AVAILABILITY_MAPPING) {
+    if (inDateRanges(run, [range])) {
+      return AVAILABILITY_MAPPING[range].label
+    }
+  }
+}
+
+export const runStartDate = (courseRun: CourseRun) =>
+  moment(courseRun.best_start_date || DEFAULT_START_DT, DATE_FORMAT)
+
+export const runEndDate = (courseRun: CourseRun) =>
+  moment(courseRun.best_end_date || DEFAULT_END_DT, DATE_FORMAT)
+
+export const compareRuns = (firstRun: CourseRun, secondRun: CourseRun) =>
+  runStartDate(firstRun).diff(runStartDate(secondRun), "hours")
+
+export const bestRun = (runs: Array<CourseRun>) => {
+  // Runs that are running right now
+  const currentRuns = runs.filter(
+    // $FlowFixMe: runStartDate and runEndDate will always return a Moment
```

Comment:
you might be able to get rid of this flow disable by adding a return type to those functions

Reply:
✔️